conslogging/conslogging_test.go

```diff
@@ -0,0 +1,85 @@
+package conslogging
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_prettyPrefix(t *testing.T) {
+	testCases := []struct {
+		name          string
+		prefixPadding int
+		prefix        string
+		expected      string
+	}{
+		{
+			name:          "does not truncate if prefixPadding is NoPadding",
+			prefixPadding: NoPadding,
+			prefix:        "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+			expected:      "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+		},
+		{
+			name:          "shortens git SHA if prefix is present",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:80524f0d82a353b3444e83f056207e15f4d5447c+hello-world",
+			expected:      "g/e/earthly:80524f+hello-world",
+		},
+		{
+			name:          "keeps branch name",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:some-feature-branch+hello-world",
+			expected:      "g/e/earthly:some-feature-branch+hello-world",
+		},
+		{
+			name:          "keeps branch name containing special characters /-_",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:-_/ryan_-/branch-names/-_in-here+hello-world",
+			expected:      "g/e/earthly:-_/ryan_-/branch-names/-_in-here+hello-world",
+		},
+		{
+			name:          "keeps branch name containing _",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:ryan/branch+hello-world",
+			expected:      "g/e/earthly:ryan/branch+hello-world",
+		},
+		{
+			name:          "keeps branch name containing -",
+			prefixPadding: DefaultPadding,
+			prefix:        "github.com/earthly/earthly:ryan/branch+hello-world",
+			expected:      "g/e/earthly:ryan/branch+hello-world",
+		},
+		{
+			name:          "target with no github info",
+			prefixPadding: DefaultPadding,
+			prefix:        "+run",
+			expected:      padding(DefaultPadding-4) + "+run",
+		},
+		{
+			name:          "does not add padding if prefix longer than prefixPadding",
+			prefixPadding: 3,
+			prefix:        "+run",
+			expected:      "+run",
+		},
+		{
+			name:          "negative padding results in no change",
+			prefixPadding: -10,
+			prefix:        "+run",
+			expected:      "+run",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			assert.Equal(t, tc.expected, prettyPrefix(tc.prefixPadding, tc.prefix))
+		})
+	}
```

Comment:
Not a huge deal in this case, but I find that always running tests in parallel (a) helps uncover bugs and (b) keeps feedback cycles very quick. These tests don't strictly need it, but as far as I can tell they shouldn't be harmed by it.
```suggestion
	for _, tc := range testCases {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()
			assert.Equal(t, tc.expected, prettyPrefix(tc.prefixPadding, tc.prefix))
		})
	}
```

Reply:
now they're in parallel.